src/dfx/src/actors/pocketic.rs

```diff
@@ -115,16 +116,30 @@ impl PocketIc {
     fn wait_for_ready(
         port_file_path: &Path,
         shutdown_signal: Receiver<()>,
+        docker_container_id: Option<String>,
     ) -> Result<u16, ControlFlow<(), DfxError>> {
         let mut retries = 0;
         loop {
-            if let Ok(content) = std::fs::read_to_string(port_file_path) {
+            if let Some(container_id) = docker_container_id.as_ref() {
+                let output = std::process::Command::new("docker")
+                    .args(["exec", container_id.as_str(), "cat", "pocket-ic-port"])
```

Comment:
Why is the (host) port retrieved from the docker container? I'd expect the host port to be stored on the host and mapped to a (fixed) port used in the docker container.